### PR TITLE
Adding example decay fragments to X>YH

### PR DESCRIPTION
**Pull request by acarvalh**
More decay modes will come, following those examples.
The samples request should be done ASAP.
Thank you for reviewing

### CODE REVIEW

**Comment by acarvalh** [comment]
Dear, I just made a last addition to the cards. Can this PR be reviewed?

**Comment by agrohsje** [comment]
sihyun and myself stepped down from this task. let me ping @Saptaparna and @GurpreetSinghChahal to make sure your pr is seen.

**Comment by acarvalh** [comment]
Hi @Saptaparna and @GurpreetSinghChahal , thank you for taking this task

**Comment by menglu21** [comment]
Hi @acarvalh, i didn't review all the cards, but since you are updating them based on old cards, i suggest you to update those cards too, e.g., the "True = pdfwgt" should be switched to "False = pdfwgt". And in the runcard, the flavor scheme is set to 4, but the updated proc only use "u d c", and i'm also wondering the purpose why you use only "u d c".

**Comment by acarvalh** [comment]
Hi @menglu21 thank you for starting the review of the cards. 
The PR should have contained only the decay fragments, sorry for my mistake. 
All the madgraph cards in this PR right now are back to the form of cms-sw:master by doing direct checkout from it right now.

**Comment by menglu21** [comment]
> Hi @menglu21 thank you for starting the review of the cards. The PR should have contained only the decay fragments, sorry for my mistake. All the madgraph cards in this PR right now are back to the form of cms-sw:master by doing direct checkout from it right now.

Hi @acarvalh , we may need some time to make sure the resonancedecayfilter is fine, since there may be a issue on that, we'll come back to you after we confirm that.

File: genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2ZTo4Q.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                         pythiaPylistVerbosity = cms.untracked.int32(1),
+                         filterEfficiency = cms.untracked.double(1.0),
+                         pythiaHepMCVerbosity = cms.untracked.bool(False),
+                         comEnergy = cms.double(13000.),
+                         PythiaParameters = cms.PSet(
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        processParameters = cms.vstring(
+            '25:onMode = off', # Turn off all H decays 
+            '25:oneChannel = 1 1 100 22 22', # H->gamma gamma
```
Comment (menglu21) [inline genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2ZTo4Q.py:25]: same question here, see above

File: genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2ZTo4Q.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                         pythiaHepMCVerbosity = cms.untracked.bool(False),
+                         comEnergy = cms.double(13000.),
+                         PythiaParameters = cms.PSet(
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        processParameters = cms.vstring(
+            '25:onMode = off', # Turn off all H decays 
+            '25:oneChannel = 1 1 100 22 22', # H->gamma gamma  
+            '35:onMode = off',
+            '35:oneChannel = 1 1 100 23 23',  # Y->ZZ
```
Comment (menglu21) [inline genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2ZTo4Q.py:27]: same question here, see above

File: genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2T_YTo2B.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    comEnergy = cms.double(13000.),
+    PythiaParameters = cms.PSet(
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        pythia8PSweightsSettingsBlock,
+        processParameters = cms.vstring(
+            '25:onMode = off', # Turn off all H decays
+            '25:oneChannel = 1 1 100 15 15', # H->gamma gamma
+            '35:onMode = off',
+            '35:oneChannel = 1 1 100 5 -5',  # Y->bb
```
Comment (menglu21) [inline genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2T_YTo2B.py:31]: same question here, see above

File: genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2T_YTo2B.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    filterEfficiency = cms.untracked.double(1.0),
+    pythiaHepMCVerbosity = cms.untracked.bool(False),
+    comEnergy = cms.double(13000.),
+    PythiaParameters = cms.PSet(
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        pythia8PSweightsSettingsBlock,
+        processParameters = cms.vstring(
+            '25:onMode = off', # Turn off all H decays
+            '25:oneChannel = 1 1 100 15 15', # H->gamma gamma
```
Comment (menglu21) [inline genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2T_YTo2B.py:29]: same question here, see above

**Review by menglu21 — COMMENTED** [review]
besides those comments, an additional warning on the final cross section, it seems the genxsanalyzer doesn't report the filter efficiency correctly when there is resonancedecayfilter, the reported number is just the LHE level (xs number in XSDB is obtained use this method), so please always take this into account.

**Comment by acarvalh** [comment]
done @menglu21 
I hope you do not mind I added other two cards to this PR, that are very similar and would follow the same comments

As the first template (without the onIfMatch) came from @atishelmanch and he tested successfully that version I tag him here

File: genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py [hunk 5]
```diff
@@ -23,23 +23,25 @@
...
         pythia8CommonSettingsBlock,
         pythia8CP5SettingsBlock,
-        pythia8PSweightsSettingsBlock,      
+        pythia8PSweightsSettingsBlock,
         processParameters = cms.vstring(
-            '25:onMode = off', # Turn off all H decays 
-            '25:oneChannel = 1 1 100 22 22', # H->gamma gamma  
+            '25:onMode = off', # Turn off all H decays
+            '25:oneChannel = 1 1 100 22 22', # H->gamma gamma
+            '25:onIfMatch = 15 -15',
```
Comment (atishelmanch) [inline genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py:30]: Hello! @acarvalh 

This line should be `25:onIfMatch = 22 22` (I think you know but probably a typo / incorrect copy paste)

Comment (acarvalh) [reply genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py]: fixed, along with other copy/paste typo

Comment (menglu21) [reply genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py]: sorry, i'm a bit confused with the update. I guess @atishelmanch was suggesting change "25:oneChannel = 1 1 100 22 22" to "25:onIfMatch = 22 22", right? And "oneChannel" in other fragments should be checked also if it's really wanted.

Comment (acarvalh) [reply genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py]: Hi @menglu21 , no, he just linked the wrong line (checked offline). He just wanted me to fix the line that defines tau decays (line 30).

Due this comment I checked again all the cards and found only one other similar typo typo, that was corrected in this same commit

Comment (atishelmanch) [reply genfragments/ThirteenTeV/Higgs/HY/ResonanceDecayFilter_example_HY_XToYH_HTo2G_YTo2WTo2Q1L1Nu.py]: Hello @menglu21 - I confirm what @acarvalh has said.

**Comment by acarvalh** [comment]
Hi @menglu21, I am sorry for pushing. 
There are 6 analysis teams that are waiting for the central production of their samples start. 
Are you happy with my additions? If yes, do you think this PR could be merged this still week ?

**Comment by menglu21** [comment]
Hi @acarvalh , sorry for the delay, i don't have further comment. let me ping others @Saptaparna @GurpreetSinghChahal @mlizzo please have a check on the cards, and please merge if you are satisfied.

**Comment by mlizzo** [comment]
Hi @menglu21 , other than checking that ResonanceFilterDecays:daughter does take the absolute value of the particle id as argument, I have nothing else to add. It's fine to me

**Comment by acarvalh** [comment]
Hi @mlizzo 

> other than checking that ResonanceFilterDecays:daughter does take the absolute value of the particle id as argument,

That was verified by checking the results of those cards. Also all the cards used for the the production for the HH case use that moduli notation (see the parent folder for HH examples)

**Pull request merged**